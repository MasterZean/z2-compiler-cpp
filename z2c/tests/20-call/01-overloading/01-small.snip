def @main() {
	const MIN: Small = -128
	test(MIN)
}

def test(b: Byte),
def test(b: Word),
def test(b: DWord),
def test(b: QWord),
def test(b: Small),
def test(b: Short),
def test(b: Int),
def test(b: Long),
def test(b: Float),
def test(b: Double),
def test(b: Char),
def test(b: PtrSize) {
	'a'
	b.class
}

def test(move b: Byte),
def test(move b: Word),
def test(move b: DWord),
def test(move b: QWord),
def test(move b: Small),
def test(move b: Short),
def test(move b: Int),
def test(move b: Long),
def test(move b: Float),
def test(move b: Double), 
def test(move b: Char), 
def test(move b: PtrSize) {
	'm'
	b.class
}


