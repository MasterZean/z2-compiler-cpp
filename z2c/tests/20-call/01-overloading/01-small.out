void test(int8 b);

void _main() {
	int8 MIN = -128;
	::test((int8)(MIN));
}

void test(uint8 b) {
	'a';
	(b, Class(2));
}

void test(uint16 b) {
	'a';
	(b, Class(4));
}

void test(uint32 b) {
	'a';
	(b, Class(6));
}

void test(uint64 b) {
	'a';
	(b, Class(8));
}

void test(int8 b) {
	'a';
	(b, Class(1));
}

void test(int16 b) {
	'a';
	(b, Class(3));
}

void test(int32 b) {
	'a';
	(b, Class(5));
}

void test(int64 b) {
	'a';
	(b, Class(7));
}

void test(float b) {
	'a';
	(b, Class(9));
}

void test(double b) {
	'a';
	(b, Class(10));
}

void test1(uint32 b) {
	'a';
	(b, Class(12));
}

void test2(size_t b) {
	'a';
	(b, Class(13));
}

void test(uint8&& b) {
	'm';
	(b, Class(2));
}

void test(uint16&& b) {
	'm';
	(b, Class(4));
}

void test(uint32&& b) {
	'm';
	(b, Class(6));
}

void test(uint64&& b) {
	'm';
	(b, Class(8));
}

void test(int8&& b) {
	'm';
	(b, Class(1));
}

void test(int16&& b) {
	'm';
	(b, Class(3));
}

void test(int32&& b) {
	'm';
	(b, Class(5));
}

void test(int64&& b) {
	'm';
	(b, Class(7));
}

void test(float&& b) {
	'm';
	(b, Class(9));
}

void test(double&& b) {
	'm';
	(b, Class(10));
}

void test3(uint32&& b) {
	'm';
	(b, Class(12));
}

void test4(size_t&& b) {
	'm';
	(b, Class(13));
}

